servlet/src/main/java/com/linecorp/armeria/server/servlet/DefaultServletContext.java

```diff
@@ -0,0 +1,548 @@
+    }
+
+    @Override
+    public InputStream getResourceAsStream(String path) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public String getRealPath(String path) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    @Nullable
+    public ServletRequestDispatcher getRequestDispatcher(String path) {
+        requireNonNull(path, "path");
+        final Pair<String, DefaultServletRegistration> pair = servletUrlMapper.getMapping(path);
+        if (pair == null) {
+            return null;
+        }
+
+        // TODO Integrate this logic into servletUrlMapper.
+        final String pathPattern = pair.getKey();
+        final String servletPath;
+        final String pathInfo;
+        if (pathPattern.endsWith("/*")) {
+            // pathPattern: "/lawn/*"
+            // path: "/lawn/index.html" then,
+            // servletPath: "/lawn"
+            // pathInfo: "/index.html"
+            servletPath = pathPattern.substring(0, pathPattern.length() - 2);
+            pathInfo = path.substring(servletPath.length());
+        } else {
+            // pathPattern starts with "*." or exact path.
+            servletPath = path;
+            pathInfo = null;
+        }
+
+        return new ServletRequestDispatcher(new ServletFilterChain(pair.getValue()), pair.getValue().getName(),
+                                            servletPath, pathInfo);
+    }
+
+    @Override
+    @Nullable
+    public ServletRequestDispatcher getNamedDispatcher(String name) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    @Nullable
+    public Servlet getServlet(String name) throws ServletException {
+        // This method is deprecated and should return null.
+        return null;
+    }
+
+    @Override
+    public Enumeration<Servlet> getServlets() {
+        // This method is deprecated and should return an empty set.
+        return Collections.enumeration(ImmutableSet.of());
+    }
+
+    @Override
+    public Enumeration<String> getServletNames() {
+        // This method is deprecated and should return an empty set.
+        return Collections.enumeration(ImmutableSet.of());
+    }
+
+    @Override
+    public void log(String message) {
+        requireNonNull(message, "message");
+        level.log(logger, message);
+    }
+
+    @Override
+    public void log(Exception exception, String message) {
+        requireNonNull(exception, "exception");
+        requireNonNull(message, "message");
+        level.log(logger, message, exception);
+    }
+
+    @Override
+    public void log(String message, Throwable throwable) {
+        requireNonNull(message, "message");
+        requireNonNull(throwable, "throwable");
+        level.log(logger, message, throwable);
+    }
+
+    @Override
+    public String getServerInfo() {
+        return ArmeriaHttpUtil.SERVER_HEADER +
+               " (JDK " + SystemInfo.javaVersion() + ';' + SystemInfo.osType().name() + ')';
```

Comment:
Is is some sort of specification to return the Java version and OS name in this string?

Reply:
Yes, refer https://github.com/apache/tomcat/blob/master/java/jakarta/servlet/ServletContext.java java doc. 
```java
    /**
     * Returns the name and version of the servlet container on which the
     * servlet is running.
     * <p>
     * The form of the returned string is
     * <i>servername</i>/<i>versionnumber</i>. For example, the JavaServer Web
     * Development Kit may return the string
     * <code>JavaServer Web Dev Kit/1.0</code>.
     * <p>
     * The servlet container may return other optional information after the
     * primary string in parentheses, for example,
     * <code>JavaServer Web Dev Kit/1.0 (JDK 1.1.6; Windows NT 4.0 x86)</code>.
     *
     * @return a <code>String</code> containing at least the servlet container
     *         name and version number
     */
```